src/v/redpanda/application.cc

```diff
@@ -1164,6 +1164,14 @@ void application::start_redpanda(::stop_signal& app_signal) {
       [group_migration] { group_migration->await().get(); });
 
     _deferred.emplace_back([this] { controller->shutdown_input().get(); });
+
+    // Started after controller due to dependency on health monitor backend
+    metadata_cache
```

Comment:
This is really complex logic related with shard, now we need to remember that metadata cache start must only be called on a single shard.

Reply:
If you forget, you'll get an instant vassert() ;-)  I could also take the approach of stop() and just add a condition in there so callers don't have to worry about it.

Reply:
Next update will allow calling start on all shards.

Reply:
New patch addresses this, please resolve this if satisfied.